src/Service/ViewHelperManagerFactory.php

```diff
@@ -41,39 +40,41 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
     {
         $options = $options ?: [];
         $options['factories'] = isset($options['factories']) ? $options['factories'] : [];
-        $plugins = parent::__invoke($container, $requestedName, $options);
+
+        $managerConfig = [];
+        if ($options) {
+            $managerConfig = $options;
+        } elseif ($container->has('config')) {
+            $managerConfig = $container->get('config')['view_helpers'] ?? [];
+        }
 
         // Override plugin factories
-        $plugins = $this->injectOverrideFactories($plugins, $container);
+        $managerConfig = $this->injectOverrideFactories($managerConfig, $container);
 
-        return $plugins;
+        return new HelperPluginManager($container, $managerConfig);
     }
 
     /**
-     * Inject override factories into the plugin manager.
-     *
-     * @param HelperPluginManager $plugins
-     * @param ContainerInterface $services
-     * @return HelperPluginManager
+     * Inject override factories into the plugin manager config.
      */
-    private function injectOverrideFactories(HelperPluginManager $plugins, ContainerInterface $services)
+    private function injectOverrideFactories(array $managerConfig, ContainerInterface $services) : array
     {
         // Configure URL view helper
         $urlFactory = $this->createUrlHelperFactory($services);
-        $plugins->setFactory(ViewHelper\Url::class, $urlFactory);
-        $plugins->setFactory('laminasviewhelperurl', $urlFactory);
+        $managerConfig['aliases']['laminasviewhelperurl'] = ViewHelper\Url::class;
```

Comment:
I think we can drop these "normalized" aliases - users should only be using the FCQN or the class-name-minus-namespace when retrieving helpers.

Reply:
That change would be outside of the scope of this PR. Too easy to lose track of small details like that during review.

I plan to drop all obsolete aliases in another PR.

Reply:
:+1: